### ISSUE TITLE
Using an an empty `ShadDecoration()` results in changes

### ISSUE THREAD

**Issue by dickermoshe** [nank1ro/flutter-shadcn-ui#188]
### Steps to reproduce

ShadAlert with an empty `ShadDecoration()`

### Expected results

No changes from default `ShadAlert`

(I think it's time to start adding widget tests 🤷)

### Actual results

The border is removed

### shadcn_ui version

0.15.0

### Platform

MacOS, Windows, Linux, Android, iOS, Web

### Code sample

<details open><summary>Code sample</summary>

```dart
class AlertPage extends StatelessWidget {
  const AlertPage({super.key});

  @override
  Widget build(BuildContext context) {
    return BaseScaffold(
      appBarTitle: 'Alert',
      children: [
        ConstrainedBox(
          constraints: const BoxConstraints(maxWidth: 600),
          child: const ShadAlert(
            iconSrc: LucideIcons.terminal,
            decoration: ShadDecoration(),
            title: Text('Heads up!'),
            description:
                Text('You can add components to your app using the cli.'),
          ),
        ),
      ],
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

<img width="1073" alt="image" src="https://github.com/user-attachments/assets/2b846377-b6f8-4b3f-8375-2b83702914bd">

</details>

### Logs

<details open><summary>Logs</summary>

```console
NA
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.24.3, on Microsoft Windows [Version 10.0.22631.4317], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.11.5)
[√] Android Studio (version 2023.1)
[√] VS Code, 64-bit edition (version 1.93.1)
[!] Proxy Configuration
    ! NO_PROXY does not contain ::1
[√] Connected device (3 available)
[√] Network resources

! Doctor found issues in 1 category.
```

</details>

**Comment by nank1ro** [comment]
This is working as intended.
The empty ShadDecoration class has all set to null and merge to true, so it's fully merged to the default one.
Use ShadDecoration.none to disable it completely or override specific properties and set merge to false to disable merging.

**Comment by nank1ro** [comment]
If you want to disable the border use ShadDecoration(border: Shard border.none)

Or use secondaryBorder or focusedBorder depending on your needs

**Comment by dickermoshe** [comment]
> The empty ShadDecoration class has all set to null and merge to true, so it's fully merged to the default one.
You misunderstand, the opposite actually happens. Everything is being set to null, so the border is being removed  See the source code:

https://github.com/nank1ro/flutter-shadcn-ui/blob/1bbe4cdc8fb559125f1d0ed18767e061ed507efe/lib/src/components/alert.dart#L91-L99

There is no merge happening here.

**Comment by nank1ro** [comment]
If you want to disable the border use ShadDecoration(border: Shard border.none)

Or use secondaryBorder or focusedBorder depending on your needs.

It's probably better if I update the documentation about it

**Comment by nank1ro** [comment]
> > The empty ShadDecoration class has all set to null and merge to true, so it's fully merged to the default one.
> You misunderstand, the opposite actually happens. Everything is being set to null, so the border is being removed  See the source code:
> 
> https://github.com/nank1ro/flutter-shadcn-ui/blob/1bbe4cdc8fb559125f1d0ed18767e061ed507efe/lib/src/components/alert.dart#L91-L99
> 
> There is no merge happening here.

You're right it's a bug.
Will add comments and tests before releasing the v1

**Comment by saladthieves** [comment]
> (I think it's time to start adding widget tests 🤷)
@dickermoshe I had opened a [discussion](https://github.com/nank1ro/flutter-shadcn-ui/discussions/189) regarding this. Let me know your thoughts :)

**Comment by nank1ro** [comment]
fixed in the v0.15.1
